src/main/java/com/redislabs/redistimeseries/CreateParams.java

```diff
@@ -0,0 +1,113 @@
+package com.redislabs.redistimeseries;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import redis.clients.jedis.Protocol;
+import redis.clients.jedis.util.SafeEncoder;
+
+public class CreateParams {
+
+  private Long retentionTime;
+  private boolean uncompressed;
+  private Long chunkSize;
+  private DuplicatePolicy duplicatePolicy;
+  private Map<String, String> labels;
+
+  public static CreateParams createParams() {
+    return new CreateParams();
+  }
+
+  public CreateParams() {}
+
+  public CreateParams retentionTime(long retentionTime) {
+    this.retentionTime = retentionTime;
+    return this;
+  }
+
+  public CreateParams uncompressed() {
+    this.uncompressed = true;
+    return this;
+  }
+
+  public CreateParams chunkSize(long chunkSize) {
+    this.chunkSize = chunkSize;
+    return this;
+  }
+
+  public CreateParams duplicatePolicy(DuplicatePolicy duplicatePolicy) {
+    this.duplicatePolicy = duplicatePolicy;
+    return this;
+  }
+
+  public CreateParams label(String label, String value) {
+    if (this.labels == null) this.labels = new HashMap<>();
+    this.labels.put(label, value);
+    return this;
+  }
+
+  public CreateParams labels(String... labels) {
+    final int len = labels.length;
+    if ((len & 1) != 0) {
+      throw new IllegalArgumentException();
+    }
+    Map<String, String> map = new HashMap<>(len >> 1);
+    for (int i = 0; i < len; i += 2) {
+      map.put(labels[i], labels[i + 1]);
+    }
+    return labels(map);
+  }
+
+  public CreateParams labels(Map<String, String> labels) {
+    this.labels = labels;
+    return this;
+  }
+
+  public byte[][] getCreateByteParams(String key) {
```

Comment:
I think it's better to drop getCreateByteParams and getAddByteParams, and make addOptionalParams public.
The caller should pass the list and this class shouldn't not deal with serializing external arguments like key, time, value

Reply:
@gkorland That approach would require more intermediate lists/arrays. For example, creating a list/array in `addOptionalParams` with all the optional arguments and then creating another list/array unifying key, time, value with previous one.

Reply:
Why? exactly like you pass `addOptionalParams(List<byte[]> params)` is getting a list now, it can get the list in `add` with the key, timestamp and value already there.

Reply:
@gkorland Let me know if prefer #66

Reply:
yes I think it's the right spread of responsibility